src/vs/editor/common/controller/cursorWordOperations.ts

```diff
@@ -438,6 +438,91 @@ export class WordOperations {
 		return new Range(lineNumber, column, position.lineNumber, position.column);
 	}
 
+
+	protected static _deleteWordEntireWhitespace(model: ICursorSimpleModel, position: Position): Range | null {
+		const lineContent = model.getLineContent(position.lineNumber);
+		const startIndex1 = position.column - 1; // deleteRight
+		const startIndex2 = position.column - 2; // deleteLeft
+		const firstNonWhitespace = this._findFirstNonWhitespaceChar(lineContent, startIndex1);
+		const lastNonWhitespace = strings.lastNonWhitespaceIndex(lineContent, startIndex2);
+		if ((startIndex1 + 1 < firstNonWhitespace && lastNonWhitespace + 1 < startIndex2) || (startIndex1 + 1 >= firstNonWhitespace && lastNonWhitespace + 1 < startIndex2) || (startIndex1 + 1 <= firstNonWhitespace && lastNonWhitespace + 1 === startIndex2)) {
+			return new Range(position.lineNumber, lastNonWhitespace + 2, position.lineNumber, firstNonWhitespace + 1);
+		}
+		return null;
+	}
+
+
+	public static deleteWordEntire(ctx: DeleteWordContext, wordNavigationType: WordNavigationType): Range | null {
+		const wordSeparators = ctx.wordSeparators;
+		const model = ctx.model;
+		const selection = ctx.selection;
+		const whitespaceHeuristics = ctx.whitespaceHeuristics;
+
+		if (!selection.isEmpty()) {
+			return selection;
+		}
+
+		if (DeleteOperations.isAutoClosingPairDelete(ctx.autoClosingBrackets, ctx.autoClosingQuotes, ctx.autoClosingPairs.autoClosingPairsOpen, ctx.model, [ctx.selection])) {
+			const position = ctx.selection.getPosition();
+			return new Range(position.lineNumber, position.column - 1, position.lineNumber, position.column + 1);
+		}
+
+		const position = new Position(selection.positionLineNumber, selection.positionColumn);
+
+		let lineNumber = position.lineNumber;
+		let column = position.column;
+		let column_end = position.column;
```

Comment:
Should that be named columnEnd?

Reply:
Certainly, 'columnEnd' is proper. I will rename it.